### PR TITLE
[CDEC-403] Remove partial field accessors from `HandlerSpec`, `InductiveValidationError`, `InvariantViolation` and `ValidationResult` data types.

### PR DESCRIPTION
**Pull request by Jimbo4350**
## Description

Remove partial field accessors from `HandlerSpec`, `InductiveValidationError`, `InvariantViolation` and `ValidationResult` data types.

## Linked issue

https://iohk.myjetbrains.com/youtrack/issue/CDEC-403

## Type of change

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [ ] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [ ] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [ ] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [ ] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.

## Testing checklist

- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: infra/src/Pos/Infra/Communication/Types/Protocol.hs [hunk 1]
```diff
@@ -186,8 +186,9 @@ buildBS :: ByteString -> B.Builder
...
 
 data HandlerSpec
-    = ConvHandler { hsReplyType :: MessageCode }
-    | UnknownHandler Word8 ByteString
+    -- | ConvHandler hsReplyType
+    = ConvHandler !MessageCode
+    | UnknownHandler !Word8 !ByteString
     deriving (Show, Generic, Eq)
 
 instance Bi HandlerSpec where
```
Comment (erikd) [inline infra/src/Pos/Infra/Communication/Types/Protocol.hs]: We don't seem to have golden/bitripping tests for `HandlerSpec`. At least not in the `infra` package.

We need that before we go changing things.

Comment (Jimbo4350) [reply infra/src/Pos/Infra/Communication/Types/Protocol.hs]: Will do, I'll create a ticket.

Comment (Jimbo4350) [reply infra/src/Pos/Infra/Communication/Types/Protocol.hs]: Tests created in https://github.com/input-output-hk/cardano-sl/pull/3268 and merged. Rebased [CDEC-403] onto develop with golden and bitripping tests still passing.

File: wallet-new/test/unit/Wallet/Inductive/Validation.hs [hunk 2]
```diff
@@ -201,15 +193,21 @@ inductiveIsValid Inductive{..} = do
 -------------------------------------------------------------------------------}
 
 instance (Hash h a, Buildable a) => Buildable (InductiveValidationError h a) where
-  build InductiveInvalidBoot{..} = bprint
+  build (InductiveInvalidBoot
+             inductiveInvalidBoot
+             inductiveInvalidError) = bprint
```
Comment (mhuesch) [inline wallet-new/test/unit/Wallet/Inductive/Validation.hs]: 👍 for removing uses of RecordWildcards. Maybe at some point we can go through and drop it from `default-extensions` / drop all uses.

**Pull request merged**